Nightly test failure, intel/19.0.5 icpc: TestTeamMDRangePolicyCTAD.cpp error #177: function "<unnamed>::TestTeamThreadMDRangeCTAD::test_ctad_inside_parallel_for"

**Describe the bug**

Nightly intel/19.0.5 builds (icpc) are struggling with the ctad updates in TestTeamMDRangePolicyCTAD.cpp:
```
14:28:43 /home/jenkins/node/workspace/kokkos_osrhel7_intel_1905_openmp_pthread_serial/kokkos/core/unit_test/TestTeamMDRangePolicyCTAD.cpp(76): error #177: function "<unnamed>::TestTeamThreadMDRangeCTAD::test_ctad_inside_parallel_for" was declared but never referenced
14:28:43     static void test_ctad_inside_parallel_for() {
14:28:43                 ^
14:28:43 
14:28:43 /home/jenkins/node/workspace/kokkos_osrhel7_intel_1905_openmp_pthread_serial/kokkos/core/unit_test/TestTeamMDRangePolicyCTAD.cpp(136): error #177: function "<unnamed>::TestTeamVectorMDRangeCTAD::test_ctad_inside_parallel_for" was declared but never referenced
14:28:43     static void test_ctad_inside_parallel_for() {
14:28:43                 ^
14:28:43 
14:28:43 /home/jenkins/node/workspace/kokkos_osrhel7_intel_1905_openmp_pthread_serial/kokkos/core/unit_test/TestTeamMDRangePolicyCTAD.cpp(196): error #177: function "<unnamed>::TestThreadVectorMDRangeCTAD::test_ctad_inside_parallel_for" was declared but never referenced
14:28:43     static void test_ctad_inside_parallel_for() {
14:28:43                 ^
14:28:43 
14:28:43 compilation aborted for /home/jenkins/node/workspace/kokkos_osrhel7_intel_1905_openmp_pthread_serial/kokkos/core/unit_test/TestTeamMDRangePolicyCTAD.cpp (code 2)
14:28:43 make[2]: *** [core/unit_test/CMakeFiles/Kokkos_CoreTestCompileOnly.dir/TestTeamMDRangePolicyCTAD.cpp.o] Error 2
```

This appears to follow merge of #6954 , though this is the list of commits since the previous passing build:

**Changes**

    Add dependabot for GitHub Actions ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905_openmp_pthread_serial/1051/changes#5071f2fd343dd608d57d7cba3de735a7d0fe5f7e) / [githubweb](https://github.com/kokkos/kokkos/commit/5071f2fd343dd608d57d7cba3de735a7d0fe5f7e))
    Added static_asserts for out of range tuple_element and get ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905_openmp_pthread_serial/1051/changes#2a7ca1a37e6ac85e4b1a4f0df19e4d156df5b68e) / [githubweb](https://github.com/kokkos/kokkos/commit/2a7ca1a37e6ac85e4b1a4f0df19e4d156df5b68e))
    Improve Kokkos Graphs (#7039) ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905_openmp_pthread_serial/1051/changes#669746ef8278cda59f7ba7d1605946f93d6fe7c4) / [githubweb](https://github.com/kokkos/kokkos/commit/669746ef8278cda59f7ba7d1605946f93d6fe7c4))
    Introduce `KOKKOS_DEDUCTION_GUIDE` macro to allow user-defined deduction guide in device code for clang compiler (#6954) ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905_openmp_pthread_serial/1051/changes#63a3cef186bb6b1e68007c4fb24f59a86b44fc4c) / [githubweb](https://github.com/kokkos/kokkos/commit/63a3cef186bb6b1e68007c4fb24f59a86b44fc4c))

**Please include the following for a minimal reproducer**

**Reproducer**:
```bash
module load sems-cmake sems-intel/19.0.5

$KOKKOS_PATH/generate_makefile.bash --with-devices=Serial  --compiler=icpc --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="17" --kokkos-path=$KOKKOS_PATH
```

--- a/core/unit_test/TestNestedReducerCTAD.cpp
+++ b/core/unit_test/TestNestedReducerCTAD.cpp
@@ -26,9 +26,10 @@ struct TestNestedReducerCTAD {
   using TeamHandle  = TeamPolicy::member_type;
 
   struct FakeComparator {
-    KOKKOS_FUNCTION bool operator()(ScalarType, ScalarType) const {
+    template <class T>
+    KOKKOS_FUNCTION bool operator()(T const&, T const&) const {
       return true;
-    };
+    }
   };
 
   template <class ValueType>
@@ -237,8 +238,8 @@ struct TestNestedReducerCTAD {
     }
   }
 
-  static void test_ctad_inside_parallel_for() {
-    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO), TestNestedReducerCTAD());
+  TestNestedReducerCTAD() {
+    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO), *this);
   }
 };
 

--- a/core/unit_test/TestTeamMDRangePolicyCTAD.cpp
+++ b/core/unit_test/TestTeamMDRangePolicyCTAD.cpp
@@ -73,9 +73,8 @@ struct TestTeamThreadMDRangeCTAD {
     }
   }
 
-  static void test_ctad_inside_parallel_for() {
-    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO),
-                         TestTeamThreadMDRangeCTAD());
+  TestTeamThreadMDRangeCTAD() {
+    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO), *this);
   }
 };
 
@@ -133,9 +132,8 @@ struct TestTeamVectorMDRangeCTAD {
     }
   }
 
-  static void test_ctad_inside_parallel_for() {
-    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO),
-                         TestTeamVectorMDRangeCTAD());
+  TestTeamVectorMDRangeCTAD() {
+    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO), *this);
   }
 };
 
@@ -193,9 +191,8 @@ struct TestThreadVectorMDRangeCTAD {
     }
   }
 
-  static void test_ctad_inside_parallel_for() {
-    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO),
-                         TestThreadVectorMDRangeCTAD());
+  TestThreadVectorMDRangeCTAD() {
+    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO), *this);
   }
 };
 

